ccc/github.py

```diff
@@ -47,6 +47,8 @@ class SessionAdapter(enum.Enum):
 
 def github_api_ctor(
     github_url: str,
+    # can't use bitwise-or for forward reference
+    es_client: typing.Optional['ccc.elasticsearch.ElasticSearchClient']=None,
```

Comment:
while this comment may be helpful to understand this specific diff / why you chose not to use the  shorthand `TypeA | None` in this case, I think this comment will be confusing if read later. also, `typing.Optional` is a fine choice - there is no reason to "justify" using it.

you should be able to write the entire typehint as a str-literal, btw (`'ESC | None'`); considering there is a default-value defined, one might also argue it would be okay to simply not declare the argument to be optional (this is fairly obvious from the presence of a default-value).

in short: please rm the comment - the code itself is fine w/ me, of course